docs/source/example_pipe.rst

```diff
@@ -0,0 +1,211 @@
+.. _example_pipe:
+.. role:: bash(code)
+   :language: bash
+
+Example 3: Kinetic energy dissipation in a pipe
+===============================================
+
+In this example, we show how to minimize the shape functional
+
+.. math::
+
+    \mathcal{J}(\Omega) = \int_\Omega \nu \nabla \mathbf{u} : \nabla \mathbf{u} \,\mathrm{d}\mathbf{x}\,,
+
+where :math:`\mathbf{u}:\mathbb{R}^2\to\mathbb{R^2}` is the velocity of an incompressible
+fluid and :math:`\nu` is the fluid viscosity.
+The fluid velocity :math:`\mathbf{u}` and the fluid pressure
+:math:`p:\mathbb{R}^2\to\mathbb{R}` satisfy the incompressible Navier-Stokes
+equations
+
+.. math::
+    :nowrap:
+
+    \begin{align*}
+    -\nu \Delta \mathbf{u} + \mathbf{u} \nabla \mathbf{u} + \nabla p &= 0 & \text{in } \Omega\,,\\
+    \operatorname{div} \mathbf{u} &= 0& \text{in } \Omega\,,\\
+    \mathbf{u} &= \mathbf{g} &\text{on } \partial\Omega\setminus \Gamma\,,\\
+    p\mathbf{n} - \nu \nabla u\cdot \mathbf{n} & = 0 &\text{on } \Gamma\,.
+    \end{align*}
+
+Here, :math:`\mathbf{g}` is given by a Poiseuille flow at the inlet and is zero on the walls of the pipe.
+The letter :math:`\Gamma` denotes the outlet.
+
+In addition to the PDE-contstraint, we enforce a volume constraint:
+the volume of the domain should remain constant during the optimization process.
+
+
+The geometry of the initial domain is described in the following
+`gmsh <http://gmsh.info/>`_ script.
+
+.. literalinclude:: ../../examples/pipe/pipe.geo
+    :linenos:
+
+The mesh can be generated typing :bash:`gmsh -2 -clscale 0.1 pipe.geo`
```

Comment:
PETSc doesn't understand the format of the latest gmsh version, so the command here and in the makefile should be 
`gmsh -2 -clscale 0.1 -format msh2 pipe.geo`

Reply:
done